wolfcrypt/src/port/kcapi/kcapi_ecc.c

```diff
@@ -465,38 +453,39 @@ int KcapiEcc_Verify(ecc_key* key, const byte* hash, word32 hashLen, byte* sig,
         }
     }
     if (ret == 0) {
-        ret = posix_memalign((void*)&sigHash_aligned, pageSz, sigLen + hashLen);
-        if (ret != 0) {
+        keySz = key->dp->size;
+    #ifdef KCAPI_USE_XMALLOC
+        buf_aligned = (byte*)XMALLOC(sigLen + hashLen, key->heap,
+            DYNAMIC_TYPE_TMP_BUFFER);
+        if (buf_aligned == NULL) {
             ret = MEMORY_E;
         }
-    }
-    if (ret == 0) {
-        keySz = key->dp->size;
-        ret = posix_memalign((void*)&outbuf, pageSz, keySz * 2);
+    #else
+        ret = posix_memalign((void*)&buf_aligned, pageSz, sigLen + hashLen);
         if (ret != 0) {
             ret = MEMORY_E;
         }
+    #endif
     }
     if (ret == 0) {
-        XMEMCPY(sigHash_aligned, sig, sigLen);
-        XMEMCPY(sigHash_aligned + sigLen, hash, hashLen);
+        XMEMCPY(buf_aligned, sig, sigLen);
+        XMEMCPY(buf_aligned + sigLen, hash, hashLen);
```

Comment:
Is the hash going to be aligned?
What about a 521 bit key?

Reply:
The KCAPI expects a buffer with `sig (keySz * 2) + hash`. This has been tested with 256, 384 and 521 on customer hardware. The normal use cases will use posix_memalign for the buffer. During testing the use of XMALLOC is also supported to reduce overall memory use.